src/components/PopUpForm/PopUpForm.tsx

```diff
@@ -0,0 +1,274 @@
+  const [postalCode, setPostalCode] = useState("");
+  const [phoneNumber, setPhoneNumber] = useState("");
+  const [website, setWebsite] = useState("");
+  const [numAvailable, setNumAvailable] = useState("");
+  const [vaccineTypeString, setVaccineTypeString] = useState(
+    "Select Vaccine Type",
+  );
+  /* eslint-disable  @typescript-eslint/no-unused-vars */
+  const [vaccineType, setVaccineType] = useState(1);
+  /* eslint-enable  @typescript-eslint/no-unused-vars */
+  const [active, setActive] = useState(false);
+
+  let UTCDate: Date;
+
+  const handleSubmit = () => {
+    let valid = true;
+    if (!name) {
+      setShouldShowInvalidName(true);
+      valid = false;
+    } else {
+      setShouldShowInvalidName(false);
+    }
+
+    if (!date) {
+      setShouldShowInvalidDate(true);
+      valid = false;
+    } else {
+      setShouldShowInvalidDate(false);
+      /* eslint-disable  @typescript-eslint/no-unused-vars */
+      UTCDate = new Date(date);
+      /* eslint-enable  @typescript-eslint/no-unused-vars */
+    }
+
+    if (!address) {
+      setShouldShowInvalidAddress(true);
+      valid = false;
+    } else {
+      setShouldShowInvalidAddress(false);
+    }
+
+    if (!city) {
+      setShouldShowInvalidCity(true);
+      valid = false;
+    } else {
+      setShouldShowInvalidCity(false);
+    }
+
+    if (!province) {
+      setShouldShowInvalidProvince(true);
+      valid = false;
+    } else {
+      setShouldShowInvalidProvince(false);
+    }
+
+    if (!postalCodeIsValid(postalCode)) {
+      setShouldShowInvalidPostal(true);
+      valid = false;
+    } else {
+      setShouldShowInvalidPostal(false);
+    }
+
+    if (!website) {
+      setShouldShowInvalidURL(true);
+      valid = false;
+    } else {
+      setShouldShowInvalidURL(false);
+    }
+
+    if (valid) {
+      // POST here
+    }
+  };
+
+  const invalidNameMessage = shouldShowInvalidName
+    ? "Name must not be empty"
+    : undefined;
+
+  const invalidDateMessage = shouldShowInvalidDate
+    ? "Date is empty. Please Select a date."
+    : undefined;
+
+  const invalidAddressMessage = shouldShowInvalidAddress
+    ? "Address must not be empty"
+    : undefined;
+
+  const invalidCityMessage = shouldShowInvalidCity
+    ? "City must not be empty"
+    : undefined;
+
+  const invalidProvinceMessage = shouldShowInvalidProvince
+    ? "Province must not be empty"
+    : undefined;
+
+  const invalidPostalCodeMessage = shouldShowInvalidPostal
+    ? "Invalid Postal"
+    : undefined;
+
+  const invalidURLMessage = shouldShowInvalidURL
+    ? "URL must not be empty"
+    : undefined;
+  const activator = (
+    <Button onClick={() => setActive(!active)} disclosure>
+      {vaccineTypeString}
+    </Button>
+  );
+  return (
+    <section aria-label="pop-up" style={{ marginBottom: "2rem" }}>
+      <Card>
+        <Banner title="Submission Warning" status="warning">
+          Once you hit submit, this will record will immediately added to the
```

Comment:
Grammar is a little off: "Once you hit submit, this record will immediately be added to the..."